lpc8xx-hal-common/src/gpio.rs

```diff
@@ -60,14 +50,14 @@ use crate::{
 /// [`Peripherals`]: ../struct.Peripherals.html
 /// [module documentation]: index.html
 pub struct GPIO<State = init_state::Enabled> {
-    pub(crate) gpio  : raw::GPIO_PORT,
-               _state: State,
+    pub(crate) gpio: raw::GPIO_PORT,
+    _state: State,
 }
 
 impl GPIO<init_state::Enabled> {
-    pub(crate) fn new(gpio: raw::GPIO_PORT) -> Self {
+    pub unsafe fn new(gpio: raw::GPIO_PORT) -> Self {
```

Comment:
Now that the constructor is public, it would be nice to have a bit of documentation. Why (not) to use it, why it's `unsafe`.

Reply:
Added a note explaining that an already enabled gpio port is assumed